bindings/python/convert.py

```diff
@@ -0,0 +1,69 @@
+import argparse
+import json
+
+import torch
+
+from huggingface_hub import CommitOperationAdd, HfApi, hf_hub_download
+from safetensors.torch import save_file
+
+
+def convert_multi(model_id):
+    filename = hf_hub_download(repo_id=model_id, filename="pytorch_model.bin.index.json")
+    with open(filename, "r") as f:
+        data = json.load(f)
+
+    filenames = set(data["weight_map"].values())
+    local_filenames = []
+    for filename in filenames:
+        cached_filename = hf_hub_download(repo_id=model_id, filename=filename)
+        loaded = torch.load(cached_filename)
+        local = filename.replace(".bin", ".safetensors")
+        save_file(loaded, local, metadata={"format": "pt"})
+        local_filenames.append(local)
+
+    api = HfApi()
+    operations = [CommitOperationAdd(path_in_repo=local, path_or_fileobj=local) for local in local_filenames]
+    api.create_commit(
+        repo_id=model_id,
+        operations=operations,
+        commit_message="Adding `safetensors` variant of this model",
+        create_pr=True,
+    )
+
+
+def convert_single(model_id):
+    filename = hf_hub_download(repo_id=model_id, filename="pytorch_model.bin")
+
+    loaded = torch.load(filename)
+    local = "model.safetensors"
+    save_file(loaded, local, metadata={"format": "pt"})
+
+    api = HfApi()
+
+    api.upload_file(
+        path_or_fileobj=local,
+        create_pr=True,
+        path_in_repo=local,
+        repo_id=model_id,
+    )
+
+
+if __name__ == "__main__":
+    DESCRIPTION = """
+    Simple utility tool to convert automatically some weights on the hub to `safetensors` format.
+    It is PyTorch exclusive for now.
+    It works by downloading the weights (PT), converting them locally, and uploading them back
+    as a PR on the hub.
+    """
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+    parser.add_argument(
+        "model_id",
+        type=str,
+        help="The name of the model on the hub to convert. E.g. `gpt2` or `facebook/wav2vec2-base-960h`",
+    )
+    args = parser.parse_args()
+    model_id = args.model_id
+    try:
+        convert_multi(model_id)
+    except Exception:
```

Comment:
Can we catch a `EntryNotFoundError` here? Other exceptions in the `convert_multi` function should be raised.

Reply:
`EntryNotFoundError` is in a private section of the library `huggingface_hub.utils._errors.EntryNotFound`.

I agree catching a specific error is better, but I hesitate to depend on something private.

@Wauplin maybe you have some ideas ? Is it ok to rely on the errors even though they are meant to be private ?

Reply:
The errors are not private, they are exposed in `hugginface_hub.utils`, see [here](https://github.com/huggingface/huggingface_hub/blob/da331399b2cb88561e4594dfec383b5ba013355f/src/huggingface_hub/utils/__init__.py#L31)

Reply:
Yep, as mentioned by @sgugger errors (like a lot of other utils) can be imported from `huggingface_hub.utils`. As soon as something is in this module, you can assume it's public and meant to be supported on the long-term. They lies in the "utils" modules just because they are "advanced" features.

Reply:
Ahah, I wrote `type(error)` and figured they were in a private module.